llvm/lib/SYCLLowerIR/LowerESIMD.cpp

```diff
@@ -1027,7 +1026,7 @@ static void translateESIMDIntrinsicCall(CallInst &CI) {
   Function *F = CI.getCalledFunction();
   StringRef MnglName = F->getName();
   const char *MnglNameCStr = MnglName.data();
```

Comment:
This variable is probably became unused

Reply:
> understand whether we can implement the same in a way which is better aligned with existing stuff in clang & llvm (and potentially more reliable: dealing with mangler/demangler is evil)

The word "better" and "evil" are highly subjective in this case :) I have the opposite opinion, for example. That's OK. I would suggest that you file an issue after we merge this patch and put your design sketches there. We definitely need more people to discuss what's the best here if there are doubts. Does this sound good to you?

> There is no way to represent any arbitrary LLVM intrinsic in SPIR-V: 

As far as I can see `--spirv-allow-unknown-intrinsics` serves pretty well for this purpose. What exactly do you mean?

> I think this patch implements a conversion from DPC++ methods/functions into LLVM IR intrinsics

right

>, but it has nothing to do with SPIR-V.

It has. This is done for the SPIRV target which means I can't use device-specific built-ins like AMDGPU built-ins, which are enabled for specific target. So SPIRV is essential here.

Actually my main doubt/objection about using clang is not really complications related to hacking built-ins mechanism for our purposes. It is the need to introduce a lot of new intrinsics derived from combinations of element type, vector length and some other parameters. Not only I'd have to introduce those, I'd also have to write lowering for each, of course.

Overall, I think this kind of debate should better happen in the dedicated issue.

Reply:
> Why so? I don't understand why emitting call to an intrinsic in CGBuiltin.cpp based on some descriptor from Builtins.td requires some TargetMachine and why it won't work for Gen.

Do you mean Builtins.def? This is target-neutral list of built-ins. It is not a good place for Gen intrinsics. Also I don't think e.g. ARM built-ins can be used with targets other than ARM.

Reply:
> This variable is probably became unused

yep, I haven't yet pushed the fixed variant, just committed you suggestions. Will do shortly